server/app/com/xsn/explorer/services/TransactionRPCService.scala

```diff
@@ -114,4 +117,28 @@ class TransactionRPCService @Inject()(
 
     result.toFuture
   }
+
+  def encodeTPOSContract(
+      tposAddressStr: String,
+      merchantAddressStr: String,
+      commission: Int,
+      signature: String
+  ): FutureApplicationResult[JsValue] = {
+    val result = for {
+      tposAddress <- addressValidator.validate(tposAddressStr).toFutureOr
+      merchantAddress <- addressValidator.validate(merchantAddressStr).toFutureOr
+      encodedTposContract <- xsnService
+        .encodeTPOSContract(
+          tposAddress = tposAddress,
+          merchantAddress = merchantAddress,
+          commission = commission,
+          signature = signature
+        )
+        .toFutureOr
+    } yield {
+      Json.obj("opreturn" -> JsString(encodedTposContract))
```

Comment:
The `opreturn` is not related to what the method does, instead, you could return `data`/`hex`/etc, something that's actually meaningful, btw, `Json.obj("opreturn" -> encodedTposContract)` likely works.